argilla-server/src/argilla_server/contexts/datasets.py

```diff
@@ -266,7 +266,7 @@ async def create_metadata_property(
         db,
         name=metadata_property_create.name,
         title=metadata_property_create.title,
-        settings=metadata_property_create.settings.dict(),
+        settings=metadata_property_create.settings.model_dump(),
```

Comment:
@frascuchon I added `model_dump` thinking that it's gonna work but after taking another look the settings is an annotated type not using `model_config = ConfigDict(from_attributes=True)` so I reverted this change in another commit.